src/server/routes/challenges/delete-challenge.ts

```diff
@@ -0,0 +1,21 @@
+import { Request, IReply } from 'hapi'
+import { ChallengeModel } from '../../models'
+import * as Boom from '../../boom'
+
+export default async function handler(req: Request, reply: IReply) {
+  const challengeId = req.params.challengeId
+
+  if (challengeId === undefined || typeof challengeId !== 'string' || challengeId.length === 0) {
+    reply(Boom.badRequest())
+    return
+  }
+
+  const challenge = await ChallengeModel.findOne({ challengeid: challengeId }).exec()
```

Comment:
OK, [`Query#findOneAndRemove`](http://mongoosejs.com/docs/api.html#query_Query-findOneAndRemove) seems to work!